src/ff/prime_field.rs

```diff
@@ -0,0 +1,307 @@
+use super::{field::BinaryField, Field};
+use serde::{Deserialize, Serialize};
+use std::{
+    fmt::Debug,
+    ops::{
+        Add, AddAssign, BitAnd, BitAndAssign, BitOr, BitOrAssign, BitXor, BitXorAssign, Mul,
+        MulAssign, Neg, Not, Sub, SubAssign,
+    },
+};
+
+macro_rules! field_impl {
+    ( $( $field:ty ),* ) => { $(
```

Comment:
I would not make this a repeated thing.  We're not implementing a bunch of fields, so repeating this for each type is fine.